### PR TITLE
Added the create todo endpoint

### PR DESCRIPTION
**Pull request by vjnvisakh-jtc**
## Description
Added the create todo endpoint. There was an issue with modelFactory in getModel() in mongo-store.ts. For any new models it was always taking the first one i.e. Account. Please comment if that was the right way to go about it or not.

## Database schema changes
Add the Todo schema

### Automated test cases added
NA

### Manual test cases run
Request
{
    "title": "Create a PR 7"
}

Response
{
    "id": "6165a6ca1777c050bbef2c06",
    "title": "Create a PR 7",
    "createdAt": "2021-10-12T15:16:26.860Z"
}

### CODE REVIEW

File: test/spec/todo/todo.spec.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  it('should create a new todo item', async () => {
+    const res = await chai
+      .request(expressApp)
+      .post('/todo')
+      .set('Content-Type', 'application/json')
+      .send({ title: 'Go to the market' });
+
+    expect(res).to.have.status(201);
+    expect(res.body).to.have.property('id');
+    expect(res.body).to.have.property('title');
```
Comment (akashlakhera) [inline test/spec/todo/todo.spec.ts:39]: Maybe also check if the todo exists in the database.

File: src/models/model-factory.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import { BaseModel } from './base-model';
 
 export type ModelFactory<T extends BaseModel> = {
-  getType(): string;
+  getType(): any;
```
Comment (akashlakhera) [inline src/models/model-factory.ts:4]: We should avoid the use of any.

**Review by akashlakhera — APPROVED** [review]
Can merge after changes.

File: test/spec/todo/todo.spec.ts (hunk 1 above)
Comment (vjnvisakh-jtc) [reply test/spec/todo/todo.spec.ts:39]: During create or delete? During create would mean we can't have duplicate title todos.

File: src/models/model-factory.ts (hunk 2 above)
Comment (vjnvisakh-jtc) [reply src/models/model-factory.ts:4]: Changing type to LooseObject

File: test/spec/todo/todo.spec.ts (hunk 1 above)
Comment (akashlakhera) [reply test/spec/todo/todo.spec.ts:39]: Why? They can have a different ID.

Comment (vjnvisakh-jtc) [reply test/spec/todo/todo.spec.ts:39]: Ok you mean to say to check if the todo has been created in the database post request.

**Pull request merged**